src/features/products/components/Filter/Filter.tsx

```diff
@@ -0,0 +1,148 @@
+    dispatch(changeCategory(category));
+  };
+
+  const handleChooseBrand = (brand: string) => () => {
+    dispatch(toggleBrands(brand));
+  };
+
+  const handleChooseStar = (star: string) => () => {
+    dispatch(toggleStars(+star));
+  };
+
+  const handleFilterReset = () => {
+    dispatch(clearAllFilters());
+  };
+
+  useEffect(() => {
+    if (showFilter) {
+      document.body.style.overflow = "hidden";
+      return () => {
+        document.body.style.overflow = "unset";
+      };
+    }
+  }, [showFilter]);
+
+  return (
+    <>
+      <div
+        className={`overlay ${showFilter ? "overlay--show" : ""}`}
+        onClick={toggleFilter}></div>
+      <aside
+        className={`filter ${showFilter ? "filter--show" : ""}`}
+        onClick={handleFilterPropagation}>
+        <div className="filter__block">
+          <h3 className="filter__title">Categories</h3>
+          <ul className="filter-categories">
+            <li
+              className={`filter-categories__item ${
+                filter.category === "" ? "filter-categories__item--active" : ""
+              }`}
+              onClick={handleChooseCategory("")}>
+              <p>All categories</p>
+              <span>{products.length}</span>
+            </li>
+            {categories.map((category, index) => (
+              <li
+                key={`filterCat-${category}-${index}`}
+                className={`filter-categories__item ${
+                  filter.category === category
+                    ? "filter-categories__item--active"
+                    : ""
+                }`}
+                onClick={handleChooseCategory(category)}>
+                <p>{category}</p>
+                <span>{categoriesObj[category].count}</span>
+              </li>
+            ))}
+          </ul>
+        </div>
+        <div className="filter__block">
+          <h3 className="filter__title">Brands</h3>
+          <ul className="filter-brands">
+            {brands.map((brand, index) => (
+              <li key={`filterbrand-${brand}-${index}`}>
+                <label className="filter__label">
+                  <input
+                    className="filter__input"
+                    type="checkbox"
+                    checked={filter.brands.includes(brand)}
+                    onChange={handleChooseBrand(brand)}
+                  />
+                  <p>{brand}</p>
+                </label>
+              </li>
+            ))}
+          </ul>
+        </div>
+        <div className="filter__block">
+          <h3 className="filter__title">Rating</h3>
+          <ul className="filter-rating">
+            {starsArr.map((starCount, index) => (
+              <li key={`filterstars-${starCount}-${index}`}>
+                <label className="filter__label">
+                  <input
+                    className="filter__input"
+                    type="checkbox"
+                    checked={filter.stars.includes(+starCount)}
```

Comment:
logic in jsx

Reply:
Fixed - create universal function that checks if item includes in array